include/tvm/ir/type_relation.h

```diff
@@ -203,7 +203,6 @@ class TypeRelationNode : public TypeConstraintNode {
 
   bool SEqualReduce(const TypeRelationNode* other, SEqualReducer equal) const {
     return
-        equal(func, other->func) &&
```

Comment:
@tqchen please check here. I think we shouldn't check `func` here as it is a callback.

Reply:
Let us keep the comparison, try to apply the env func related fix here:

https://github.com/apache/incubator-tvm/pull/5160/commits/6301f214bb0f4319e7ca4bb5a2e7b3deb0a6d441